### ISSUE TITLE
Support for systemDatastoresAuthMode in azurerm_machine_learning_workspace

### ISSUE THREAD

**Issue by wojciech-jakubowski** [hashicorp/terraform-provider-azurerm#22150]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As of ARM API version 2022-12-01-preview AzureML has support for connecting to workspace storage account based on workspace's managed identity instead of account key. 

On Azure portal UI they call this setting **Storage account access type**. Under the hood its controlled by ARM json property called: systemDatastoresAuthMode (https://learn.microsoft.com/en-us/azure/templates/microsoft.machinelearningservices/2022-12-01-preview/workspaces?pivots=deployment-language-terraform#workspaceproperties-2). 

It would be nice to have support for this in AzureRM provider in azurerm_machine_learning_workspace, so it would be possible to disable SA key usage (which is not very secure).

### New or Affected Resource(s)/Data Source(s)

azurerm_machine_learning_workspace

### Potential Terraform Configuration

```hcl
resource "azurerm_machine_learning_workspace" "example" {
  name                    = "example-workspace"
  location                = azurerm_resource_group.example.location
  resource_group_name     = azurerm_resource_group.example.name
  application_insights_id = azurerm_application_insights.example.id
  key_vault_id            = azurerm_key_vault.example.id
  storage_account_id      = azurerm_storage_account.example.id
  system_datastores_auth_mode = "identity"  #or "account_key"

  identity {
    type = "SystemAssigned"
  }
}
```

**Comment by marvinbuss** [comment]
Even with this new property, you would not be able to disable key access on the connected storage account. Azure ML relies on Azure Files and uses the storage account key for mounting the file share when using compute instances. This property just ensures that the default datastores (e.g. `workspaceartifactstore`, `workspaceblobstore`) use identity-based auth.

**Comment by wojciech-jakubowski** [comment]
Yes, that is true. The networking design of AzureML is unfortunate in some places. I'd really like to get rid of storage account keys where I can.

**Comment by marvinbuss** [comment]
I fully understand the need to disable key based auth on all storage accounts. This is not an Azure ML related issue but more a limit of Azure Files which Azure ML depends on for some scenarios.

**Comment by caridinL6** [comment]
Hey folks, just stumbled upon this one. Thinking this one through, why not give the Managed Identity associated with the ML workspace access to [Azure Storage Account File Data Contributor role](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles/storage#storage-file-data-privileged-contributor) so it can also access the Files without using the account keys or am I glossing over some detail?

**Comment by marvinbuss** [comment]
Yes, this is now supported in Preview: https://learn.microsoft.com/en-us/azure/machine-learning/how-to-disable-local-auth-storage?view=azureml-api-2&tabs=portal

**Comment by midokate** [comment]
Can the option (system_datastores_auth_mode = "identity ) be added now that disabling access keys  is supported by azure ml ? or there is an impediment

**Comment by sivakrishnareddysunkara** [comment]
looks like system_datastores_auth_mode = "identity " is not added in the configuration is not updated in the latest version. i have tried using Azapi latest version 2025-09-01 but looks like its not stable and throwing below error
 --------------------------------------------------------------------------------
│ RESPONSE 200: 200 OK
│ ERROR CODE: InternalServerError
│ --------------------------------------------------------------------------------
│ {
│   "status": "Failed",
│   "error": {
│     "code": "InternalServerError",
│     "message": "InternalServerError"
│   }
│ }


below is my configuration


resource "azapi_resource" "aml" {
  type      = "Microsoft.MachineLearningServices/workspaces@2025-09-01"
  parent_id = azurerm_resource_group.main.id
  name      = var.aml_name
  location  = var.location
  identity {
    type         = "SystemAssigned, UserAssigned"
    identity_ids = [azurerm_user_assigned_identity.this.id]
  }
  body = {
    properties = {
      applicationInsights = azurerm_application_insights.this.id
      keyVault            = azurerm_key_vault.aml_kv.id
      publicNetworkAccess = "Disabled"
      storageAccount      = azurerm_storage_account.this.id
      v1LegacyMode        = false
      enableServiceSideCMKEncryption = true
      encryption = {
        identity = {
          userAssignedIdentity = azurerm_user_assigned_identity.this.id
        }
        keyVaultProperties = {
          keyIdentifier = azurerm_key_vault_key.cmk.id
          keyVaultArmId = azurerm_key_vault.aml_kv.id
        }
        status = "Enabled"
      }
      systemDatastoresAuthMode = "Identity"
      v1LegacyMode = false
      managedNetwork = {
        enableFirewallLog = false
        enableNetworkMonitor = false
        enbleprivateendpointconnections = true
        #firewallPublicIpAddress = "string"
        primaryUserAssignedIdentity: azurerm_user_assigned_identity.this.id
        firewallSku = "Standard"
        isolationMode = "Disabled"
        managedNetworkKind = "V2"
        # outboundRules = {
        #   {customized property} = {
        #     category = "string"
        #     status = "string"
        #     type = "string"
        #     // For remaining properties, see OutboundRule objects
        #   }
        # }
        status = {
          sparkReady = true
          status = "Active"
        }
      }
    }
    sku = {
      name = "Basic"
      tier = "Basic"
    }
  }
  ignore_casing             = true
  schema_validation_enabled = false
  response_export_values    = ["*"]
}



@marvinbuss can you please advise me how to mitigate this issue
